FlowCrypt/Controllers/Threads/ThreadDetailsViewController.swift

```diff
@@ -304,90 +422,101 @@ extension ThreadDetailsViewController {
         hideSpinner()
 
         let messageIndex = indexPath.section - 1
-        let isAlreadyProcessed = input[messageIndex].processedMessage != nil
+        let isAlreadyProcessed = messageIndex < input.count && input[messageIndex].processedMessage != nil
 
         if !isAlreadyProcessed {
-            input[messageIndex].processedMessage = processedMessage
-            input[messageIndex].isExpanded = true
+            if messageIndex < input.count {
+                input[messageIndex].rawMessage = processedMessage.message
+                input[messageIndex].processedMessage = processedMessage
+                input[messageIndex].isExpanded = true
+            } else {
+                input.append(Input(message: processedMessage.message, isExpanded: true, processedMessage: processedMessage))
+            }
 
             UIView.animate(
                 withDuration: 0.2,
                 animations: {
-                    self.node.reloadSections(IndexSet(integer: indexPath.section), with: .fade)
+                    if indexPath.section < self.node.numberOfSections {
+                        self.node.reloadSections([indexPath.section], with: .automatic)
+                    } else {
+                        self.node.insertSections([indexPath.section], with: .automatic)
+                    }
                 },
                 completion: { [weak self] _ in
                     self?.node.scrollToRow(at: indexPath, at: .middle, animated: true)
+                    self?.decryptDrafts()
                 })
         } else {
-            input[messageIndex].processedMessage?.signature = processedMessage.signature
-            node.reloadSections(IndexSet(integer: indexPath.section), with: .fade)
+            input[messageIndex].rawMessage = processedMessage.message
+            input[messageIndex].processedMessage = processedMessage
+            node.reloadSections([indexPath.section], with: .automatic)
         }
     }
 
-    private func handleError(_ error: Error, at indexPath: IndexPath) {
+    private func handle(error: Error, at indexPath: IndexPath) {
         logger.logInfo("Error \(error)")
         hideSpinner()
 
         switch error as? MessageServiceError {
-        case let .missingPassPhrase(message):
-            handleWrongPassPhrase(for: message, at: indexPath)
+        case .missingPassPhrase:
+            handleWrongPassPhrase(indexPath: indexPath)
         default:
             // TODO: - Ticket - Improve error handling for ThreadDetailsViewController
             if let someError = error as NSError?, someError.code == Imap.Err.fetch.rawValue {
                 // todo - the missing msg should be removed from the list in inbox view
                 // reproduce: 1) load inbox 2) move msg to trash on another email client 3) open trashed message in inbox
-                showToast("Message not found in folder: \(thread.path)")
+                showToast("Message not found in folder: \(inboxItem.folderPath)")
```

Comment:
It seems it's not a localized text. Am I right?

Reply:
Yep, there are still some non-localized string in the app, I'll add localization here.